core/src/main/scala/org/apache/spark/scheduler/SchedulerBackend.scala

```diff
@@ -41,4 +41,11 @@ private[spark] trait SchedulerBackend {
    */
   def applicationId(): String = appId
 
+  /**
+   * Get an application ID associated with the job.
+   *
+   * @return An application attempt id
+   */
+  def applicationAttemptId(): String = ""
```

Comment:
Yeah, I went back and forth on this bug forgot to revisit the code. I agree that `Option` would be better here, let me see how big a change that would be.